salalib/axialminimiser.cpp

```diff
@@ -253,36 +257,46 @@ void AxialMinimiser::fewestLongest(std::map<int,pvecint>& axsegcuts, std::map<Ra
 
 ///////////////////////////////////////////////////////////////////////////////////////////
 
-bool AxialMinimiser::checkVital(int checkindex, pvecint& axsegcuts, std::map<RadialKey,RadialSegment>& radialsegs, std::map<RadialKey, pvecint> &rlds, pqvector<RadialLine>& radial_lines)
+bool AxialMinimiser::checkVital(int checkindex, std::set<int> &axSegCut, std::map<RadialKey,RadialSegment>& radialsegs,
+                                std::map<RadialKey, std::set<int> > &rlds, std::vector<RadialLine>& radial_lines)
 {
    std::map<int,SalaShape>& axiallines = m_alllinemap->m_shapes;
 
    bool presumedvital = true;
    int nonvitalcount = 0, vitalsegs = 0;
    // again, this time more rigourously... check any connected pairs don't cover the link...
-   for (size_t k = 0; k < axsegcuts.size(); k++) {
-      if (m_radialsegcounts[axsegcuts[k]] <= 1) {
+   for (int cut: axSegCut) {
+      if (m_radialsegcounts[cut] <= 1) {
          bool nonvitalseg = false;
          vitalsegs++;
-         auto radialSegIter = depthmapX::getMapAtIndex(radialsegs, axsegcuts[k]);
+         auto radialSegIter = depthmapX::getMapAtIndex(radialsegs, cut);
          const RadialKey& key = radialSegIter->first;
          RadialSegment& seg = radialSegIter->second;
-         pvecint& divisorsa = rlds.find(key)->second;
-         pvecint& divisorsb = rlds.find(seg.radial_b)->second;
-         RadialLine& rlinea = radial_lines.search(key);
-         RadialLine& rlineb = radial_lines.search(seg.radial_b);
-         for (size_t divi = 0; divi < divisorsa.size(); divi++) {
-            if (divisorsa[divi] == checkindex || m_removed[divisorsa[divi]]) {
+         std::set<int>& divisorsa = rlds.find(key)->second;
+         std::set<int>& divisorsb = rlds.find(seg.radial_b)->second;
+         auto iterKey = std::find(radial_lines.begin(), radial_lines.end(), key);
+         if(iterKey == radial_lines.end()) {
+             throw depthmapX::RuntimeException("Out of range");
```

Comment:
Better exception? At least a mention what is out of range? Is it even out of range, or is something else in the data messed up?